src/app/(proper_react)/redesign/(authenticated)/user/welcome/View.tsx

```diff
@@ -41,6 +42,25 @@ export const View = ({
   const skipInitialStep = stepId === "enterInfo";
   const [currentStep, setCurrentStep] = useState<StepId>(stepId);
   const router = useRouter();
+  const record = useTelemetry();
+
+  useEffect(() => {
+    let pageName = "welcome";
+    if (currentStep === "enterInfo") {
+      pageName = "enter_scan_info";
+    } else if (currentStep === "findExposures") {
+      pageName = "scanning_for_exposures";
+    }
+
+    record("page", "view", {
+      utm_campaign: "broker_scan",
+      utm_content: pageName,
+      utm_term: skipInitialStep ? "legacy_user" : "new_user",
+    });
+
+    // Only capture telemetry on step changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I don't think this is right - `skipInitialStep` or `record` _shouldn't_ change, so the effect should still only run once. If we do see it running multiple times, there's probably something going wrong that we don't want to suppress, but instead want to fix.

```suggestion
```

Reply:
That’s right! I added all the `useEffect` depencies in [a1896c8](https://github.com/mozilla/blurts-server/pull/3823/commits/a1896c84fb7fa21e087ed6e660f45e474a694ec1). Tested this part locally running `dev` which fired one too many (because we are running Strict Mode in React 18) and I’ve mistakenly believed this was caused by one of those dependencies.